Admin/AclAdminExtension.php

```diff
@@ -42,16 +43,22 @@ class AclAdminExtension extends AdminExtension
     protected $roleHierarchy;
 
     /**
-     * @param SecurityContextInterface $securityContext
-     * @param Connection               $databaseConnection
-     * @param array                    $roleHierarchy
+     * @param SecurityContextInterface|TokenStorageInterface $tokenStorage
+     * @param Connection                                     $databaseConnection
+     * @param array                                          $roleHierarchy
      */
     public function __construct(
-        SecurityContextInterface $securityContext,
+        $tokenStorage,
         Connection $databaseConnection,
         array $roleHierarchy = array()
     ) {
-        $this->securityContext = $securityContext;
+        if (!$tokenStorage instanceof TokenStorageInterface && !$tokenStorage instanceof SecurityContextInterface) {
+            throw new \InvalidArgumentException('$securityContext must be an instanse of SecurityContextInterface or of TokenInterface '
+                    . 'but [' .  get_class($tokenStorage) . '] given.');
```

Comment:
>      remove the . 'but [' . get_class($tokenStorage) . '] given.' part (E_WARNING if not a class)
>     use the long class name Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface and Symfony\Component\Security\Core\SecurityContextInterface
>     fix the typo instance instead of instanse
> 
@meyerbaptiste yes, I  understood you earlyer. but may be there are some problems with push may be...hm. Now done.